presto-tpch/src/test/java/com/facebook/presto/tpch/statistics/TableStatisticsRecorder.java

```diff
@@ -18,48 +18,86 @@
 import io.airlift.tpch.TpchEntity;
 import io.airlift.tpch.TpchTable;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import static com.google.common.collect.ImmutableMap.toImmutableMap;
 import static java.lang.String.format;
-import static java.util.function.Function.identity;
 
 class TableStatisticsRecorder
 {
     <E extends TpchEntity> TableStatisticsData recordStatistics(TpchTable<E> tpchTable, Predicate<E> constraint, double scaleFactor)
     {
-        Iterable<E> rows = tpchTable.createGenerator(scaleFactor, 1, 1);
-        return recordStatistics(rows, tpchTable.getColumns(), constraint);
+        int parallelPartsToProcess = Runtime.getRuntime().availableProcessors();
+        if (tpchTable.equals(TpchTable.NATION) || tpchTable.equals(TpchTable.REGION)) {
+            //These tables have too few rows to benefit from parallel processing
+            parallelPartsToProcess = 1;
+        }
+        final ArrayList<CompletableFuture<TablePartStatistics>> statsRecorders = new ArrayList<>();
```

Comment:
We wait for all the `CompletableFuture`'s to finish (Line 54), then merge them; this can happen in any order, it doesn't impact correctness in any way